src/app/home-page/components/slideshow/slideshow.component.ts

```diff
@@ -0,0 +1,205 @@
+    clearInterval(this.intervalStart);
+  }
+
+  private moveSlide(direction: SLIDE_DIRECTION): void {
+    this.allSlides.nativeElement.style.transition = '';
+    // default values, that matches LEFT direction to reduce code
+    let counter = 0;
+    let translatePostion = this.translateStep;
+    let step = this.currentTranslatePosition + this.translateStep;
+    let commonCounterAction = 'dec';
+    let itemToMove = this.lastItem;
+    let itemToMovePosition = -this.totalSlidesSize;
+    let counterAction = this.slidesLength - 1;
+
+    if (direction === SLIDE_DIRECTION.RIGHT) {
+      counter = this.slidesLength - 1;
+      translatePostion = -this.translateStep;
+      step = -100;
+      commonCounterAction = 'inc';
+      itemToMove = this.firstItem;
+      itemToMovePosition = this.totalSlidesSize;
+      counterAction = 0;
+    }
+
+    if (this.counter === counter) {
+      this.handleLastAndFirstMoves(step, itemToMove, itemToMovePosition, counterAction);
+    } else {
+      this.commonSlidesMove(translatePostion, commonCounterAction);
+    }
+
+    this.selected = this.counter;
+  }
+
+  private handleClick(arr) {
+    this.subscriptions = arr.map(({ el, callback }) => {
+      const stream = fromEvent(el, 'click');
+      return stream.pipe(throttleTime(400)).subscribe(callback);
+    });
+  }
+
+  private translateItem(item, step: number): void {
+    if (item.nativeElement) {
+      item.nativeElement.style.transform = `translate(${step}%)`;
+    } else {
+      item.style.transform = `translate(${step}%)`;
+    }
+  }
+
+  private moveContainerAndItem(itemToMove: HTMLElement, direction): void {
+    this.translateItem(this.allSlides, this.currentTranslatePosition);
+    this.translateItem(itemToMove, direction);
+  }
+
+  private moveToEdgeSlideWithoutRewind(): void {
+    this.allSlides.nativeElement.style.transition = 'none';
+    let itemToMove;
+    let containerMoveDirection = 0;
+    let itemMoveDirection = 0;
+    let changeTranslatePosition = 0;
+
+    if (this.counter === 0) {
+      itemToMove = this.firstItem;
+    } else {
+      itemToMove = this.lastItem;
+      containerMoveDirection = this.translateStep - 100;
+      changeTranslatePosition = -100 + this.translateStep;
+      itemMoveDirection = 0;
+    }
+
+    this.translateItem(this.allSlides, containerMoveDirection);
+    this.translateItem(itemToMove, itemMoveDirection);
+    this.currentTranslatePosition = changeTranslatePosition;
+
+    this.allSlides.nativeElement.removeEventListener(
+      'transitionend',
+      this.moveToEdgeSlideWithoutRewind
+    );
+  }
+
+  private handleLastAndFirstMoves(changeSlidesPos, itemToMove, positionToMove, counterAction): void {
+    this.currentTranslatePosition = changeSlidesPos;
+    this.moveContainerAndItem(itemToMove, positionToMove);
+    this.allSlides.nativeElement.addEventListener(
+      'transitionend',
+      this.moveToEdgeSlideWithoutRewind
+    );
+    this.counter = counterAction;
```

Comment:
pass the counter value, not dec/inc string

Reply:
done